src/modules/headset/oculus.c

```diff
@@ -308,7 +312,18 @@ static bool oculus_getAxis(Device device, DeviceAxis axis, vec3 value) {
 }
 
 static bool oculus_vibrate(Device device, float strength, float duration, float frequency) {
-  return false; // TODO
+  int idx = device == DEVICE_HAND_LEFT ? 0 : 1;
+  state.hapticStrength[idx] = strength > 0.0f ? (strength <= 1.0f ? strength : 1.0f) : 0.0f;
+  state.hapticDuration[idx] = duration > 0.0f ? duration : 0.0f;
+  float freq = frequency / 320.0f; // 1.0 = 320hz, limit on Rift CV1 touch controllers.
+  if (freq < 0.0f) {
+    freq = 0.0f;
+  }
+  if (freq > 1.0f) {
+    freq = 1.0f;
+  }
```

Comment:
If you want, there are MIN, MAX, and CLAMP macros in util, should already be available in this file.

Reply:
done